packages/dropdown-react/src/Dropdown.tsx

```diff
@@ -78,7 +87,7 @@ export function Dropdown({
         e.target.button.focus();
         e.target.value = e.detail;
         setSelectedValue(e.detail.textContent);
-        onChange && onChange(e.detail.textContent);
+        onChange && onChange(selectedValue || "");
```

Comment:
Can we end up with a race condition here? Perhaps safer to use e.detail.textContent.

```
const nextSelectedValue = e.detail.textContent;
setSelectedValue(nextSelectedValue)
onChange(nextSelectedValue || "")
```

Reply:
I was going to say that `e.detail.textValue` is the wrong value here, which it is, but then I saw that it is used as it is now as well 😑 We want to return the `value` from the value pair, but the text content is the `label`. I'll take a look at getting this right.